src/components/artsy.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from "react"
+import * as sharify from "sharify"
+
+/**
+ * The currently signed-in user.
+ */
+export interface CurrentUser extends sharify.CurrentUser {}
+
+/**
+ * The Artsy specific props injected by the higher-order component produced by `ContextConsumer`.
+ *
+ * @see {@link ContextProvider}
+ * @see {@link ContextConsumer}
+ *
+ * @todo Once this PR for `rest` https://github.com/Microsoft/TypeScript/pull/13470 lands we’ll be able to not make this
+ *       optional and simply remove it from the props that a component wrapped with the `ContextConsumer` HOC accepts.
+ */
+export interface ContextProps {
+  artsy?: Props
```

Comment:
Oh, I remember now that there was 1 technical reason to have this prop. Without it this precondition (which I figured would be helpful during dev) wouldn’t be able to be checked https://github.com/artsy/reaction-force/pull/142/files#diff-0b367d006984fb3e2ec7231fbb80560aR76.

Does that change your thoughts?